bldsys/cmake/template/sample_api/sample.cpp.in

```diff
@@ -0,0 +1,174 @@
+	}
+}
+
+bool @SAMPLE_NAME@::prepare(vkb::Platform &platform)
+{
+	if (!ApiVulkanSample::prepare(platform))
+	{
+		return false;
+	}
+
+	prepare_pipelines();
+	build_command_buffers();
+	prepared = true;
+	return true;
+}
+
+void @SAMPLE_NAME@::prepare_pipelines()
+{
+	// Create a blank pipeline layout.
+	// We are not binding any resources to the pipeline in this sample.
+	VkPipelineLayoutCreateInfo layout_info = vkb::initializers::pipeline_layout_create_info(nullptr, 0);
+	VK_CHECK(vkCreatePipelineLayout(get_device().get_handle(), &layout_info, nullptr, &sample_pipeline_layout));
+
+	VkPipelineVertexInputStateCreateInfo vertex_input = vkb::initializers::pipeline_vertex_input_state_create_info();
+
+	// Specify we will use triangle lists to draw geometry.
+	VkPipelineInputAssemblyStateCreateInfo input_assembly = vkb::initializers::pipeline_input_assembly_state_create_info(VK_PRIMITIVE_TOPOLOGY_TRIANGLE_LIST, 0, VK_FALSE);
+
+	// Specify rasterization state.
+	VkPipelineRasterizationStateCreateInfo raster = vkb::initializers::pipeline_rasterization_state_create_info(VK_POLYGON_MODE_FILL, VK_CULL_MODE_BACK_BIT, VK_FRONT_FACE_CLOCKWISE);
+
+	// Our attachment will write to all color channels, but no blending is enabled.
+	VkPipelineColorBlendAttachmentState blend_attachment = vkb::initializers::pipeline_color_blend_attachment_state(VK_COLOR_COMPONENT_R_BIT | VK_COLOR_COMPONENT_G_BIT | VK_COLOR_COMPONENT_B_BIT | VK_COLOR_COMPONENT_A_BIT, VK_FALSE);
+
+	VkPipelineColorBlendStateCreateInfo blend = vkb::initializers::pipeline_color_blend_state_create_info(1, &blend_attachment);
+
+	// We will have one viewport and scissor box.
+	VkPipelineViewportStateCreateInfo viewport = vkb::initializers::pipeline_viewport_state_create_info(1, 1);
+
+	// Disable all depth testing.
+	VkPipelineDepthStencilStateCreateInfo depth_stencil = vkb::initializers::pipeline_depth_stencil_state_create_info(VK_FALSE, VK_FALSE, VK_COMPARE_OP_NEVER);
+
+	// No multisampling.
+	VkPipelineMultisampleStateCreateInfo multisample = vkb::initializers::pipeline_multisample_state_create_info(VK_SAMPLE_COUNT_1_BIT);
+
+	// Specify that these states will be dynamic, i.e. not part of pipeline state object.
+	std::array<VkDynamicState, 2>    dynamics{VK_DYNAMIC_STATE_VIEWPORT, VK_DYNAMIC_STATE_SCISSOR};
+	VkPipelineDynamicStateCreateInfo dynamic = vkb::initializers::pipeline_dynamic_state_create_info(dynamics.data(), vkb::to_u32(dynamics.size()));
+
+	// Load our SPIR-V shaders.
+	std::array<VkPipelineShaderStageCreateInfo, 2> shader_stages{};
+
+	// Vertex stage of the pipeline
+	shader_stages[0] = load_shader("triangle.vert", VK_SHADER_STAGE_VERTEX_BIT);
+	shader_stages[1] = load_shader("triangle.frag", VK_SHADER_STAGE_FRAGMENT_BIT);
+
+	// We need to specify the pipeline layout and the render pass description up front as well.
+	VkGraphicsPipelineCreateInfo pipe = vkb::initializers::pipeline_create_info(sample_pipeline_layout, render_pass);
```

Comment:
The naming here could be better. We use the same names in `hello_triangle` but maybe we should do better in the template. I'd e.g. call this `pipeline_create_info` instead of `pipe` to make it clear what this struct does. Same for the previous structures.